lib/Cucumber.j

```diff
@@ -881,103 +894,132 @@ function dumpGuiObject(obj)
+        mouseType = params[1],
         window = [CPApp keyWindow];
 
-    CPLog.debug("Cucapp is about to simulate a right click on the point : " + point);
+    if (mouseType != CPLeftMouseDown && mouseType != CPRightMouseDown)
+        return '{"result" : "INVALID MOUSE TYPE"}';
 
-    [self _perfomMouseEventOnPoint:point toPoint:nil window:window eventType:CPRightMouseDown numberOfClick:1 modifierFlags:params[0]];
+    CPLog.debug("Cucapp is about to simulate a mouse down to the point " + locationWindowPoint);
+
+    [self _dispatchMouseEventWithType:mouseType location:locationWindowPoint modifierFlags:flags clickCount:0 window:window];
+
+    return '{"result" : "OK"}';
 }
 
-- (CPString)simulateRightClick:(CPArray)params
+- (void)simulateMouseDown:(CPArray)params
 {
     var obj = cucumber_objects[params.shift()],
-        locationWindowPoint;
+        locationWindowPoint,
+        flags = params[0],
+        mouseType = params[1],
+        window = [CPApp keyWindow];
 
     if (!obj)
         return '{"result" : "OBJECT NOT FOUND"}';
 
+    if (mouseType != CPLeftMouseDown && mouseType != CPRightMouseDown)
+        return '{"result" : "INVALID MOUSE TYPE"}';
+
     if ([obj superview])
         locationWindowPoint = [[obj superview] convertPointToBase:CGPointMake(CGRectGetMidX([obj frame]), CGRectGetMidY([obj frame]))];
     else
         locationWindowPoint = CGPointMake(CGRectGetMidX([obj frame]), CGRectGetMidY([obj frame]));
 
-    CPLog.debug("Cucapp is about to simulate a right click on the view : " + obj);
+    CPLog.debug("Cucapp is about to simulate a mouse down on the view : " + obj);
 
-    [self _perfomMouseEventOnPoint:locationWindowPoint toPoint:nil window:[obj window] eventType:CPRightMouseDown numberOfClick:1 modifierFlags:params[0]];
+    [self _dispatchMouseEventWithType:mouseType location:locationWindowPoint modifierFlags:flags clickCount:0 window:window];
 
     return '{"result" : "OK"}';
 }
 
-- (void)simulateDoubleClickOnPoint:(CPArray)params
+- (void)simulateMouseUpOnPoint:(CPArray)params
 {
-    var point = CGPointMake(params.shift(), params.shift()),
+    var locationWindowPoint = CGPointMake(params.shift(), params.shift()),
+        flags = params[0],
+        mouseType = params[1],
         window = [CPApp keyWindow];
 
-    CPLog.debug("Cucapp is about to simulate a double click on the point : " + point);
+    if (mouseType != CPLeftMouseUp && mouseType != CPRightMouseUp)
+        return '{"result" : "INVALID MOUSE TYPE"}';
+
+    CPLog.debug("Cucapp is about to simulate a mouse up to the point " + locationWindowPoint);
+
+    [self _dispatchMouseEventWithType:mouseType location:locationWindowPoint modifierFlags:flags clickCount:0 window:window];
 
-    [self _perfomMouseEventOnPoint:point toPoint:nil window:window eventType:CPLeftMouseDown numberOfClick:2 modifierFlags:params[0]];
+    return '{"result" : "OK"}';
 }
 
-- (CPString)simulateDoubleClick:(CPArray)params
+- (void)simulateMouseUp:(CPArray)params
```

Comment:
The double log is not a problem actually, we will get information on which view we are about to click and then what is the point for this view. It's actually a better log.

By refactoring, you can avoid double code as 

``` if (mouseType != CPLeftMouseUp && mouseType != CPRightMouseUp)
        return '{"result" : "INVALID MOUSE TYPE"}';
```